aptos-move/framework/aptos-experimental/sources/trading/market/pre_cancellation_tracker.move

```diff
@@ -0,0 +1,227 @@
+module aptos_experimental::pre_cancellation_tracker {
+    use std::signer;
+    use aptos_std::big_ordered_map;
+    use aptos_std::big_ordered_map::BigOrderedMap;
+    use aptos_experimental::order_book_types::{
+        AccountClientOrderId,
+        new_account_client_order_id
+    };
+    #[test_only]
+    use std::vector;
+    #[test_only]
+    use aptos_framework::timestamp;
+
+    const DUPLICATE_ORDER_PLACEMENT: u64 = 1;
+
+    const MAX_ORDERS_GARBAGE_COLLECTED_PER_CALL: u64 = 10;
+
+    struct PreCancellationTracker has store {
+        pre_cancellation_window_micros: u64,
+        // Map of order IDs with expiration time to a boolean indicating if the order is active.
+        order_ids_with_expiration: BigOrderedMap<OrderIdWithExpiration, bool>,
```

Comment:
I found this name odd, for the purposes it does.
either order_ids_by_expiration or expiration_with_order_ids, and name the struct ExpirationAndOrderId, so that it is clear it's sorted by expiration first.

Reply:
renamed